### DISCUSSION TITLE
Overlay-Window additions

### DISCUSSION THREAD

**Discussion by Nerixyz** [Chatterino/chatterino2#5633] [Ideas]
https://github.com/Chatterino/chatterino2/pull/4746 added the initial overlay window. It's not finished yet (as hinted at in the PR). Some users might want specific features for their use-case. This discussion should be used to aggregate and discuss features.

If you have any addition you'd like to see with the overlay, please post it here. Similarly, if you know that some streamer wants a feature, post it here as well. Please include use-cases in your posts.

**Comment by Nerixyz** [comment]
The overlay should be able to scale independently of the other Windows. I don't think much discussion is needed here. There's a similar issue for popup windows (https://github.com/Chatterino/chatterino2/issues/1844).

**Reply by Nerixyz** [reply]
This is implemented in 2.5.3-beta.1 now.

**Comment by Nerixyz** [comment]
**More Control**

This is a bit generic.

- There could be a minimize button next to the close button
- There should be a hotkey for closing an overlay window (similar to the inertia shortcut)

**Comment by Zarh** [comment]
Nice feature.

bug ?  = Ctrl+shift+U works fine to make it untargettable but I doesn't work to make it targettable. I must alt+tab to get the focus on the overlay and then it's working.

Idea = make a shortcut to write a message without making it targetable if possible or at least when the message is sent. For exemple, I'm playing a game with the overlay I see a message on the overlay, I press ctrl+shilt+I to answer, I write my messge press enter and go back to my game directly.

Thank you.

**Reply by Nerixyz** [reply]
> bug ? = Ctrl+shift+U works fine to make it untargettable but I doesn't work to make it targettable. I must alt+tab to get the focus on the overlay and then it's working.

This was the initial behavior in https://github.com/Chatterino/chatterino2/pull/4746 (a global shortcut would be registered). But, this was later removed (https://github.com/Chatterino/chatterino2/pull/4746#issuecomment-2336681952).

> Idea = make a shortcut to write a message without making it targetable if possible or at least when the message is sent. For exemple, I'm playing a game with the overlay I see a message on the overlay, I press ctrl+shilt+I to answer, I write my messge press enter and go back to my game directly.

That's a good use-case - I can see this being useful. This would also require a global shortcut.

**Reply by 8thony** [reply]
Global shortcuts are also useful for pause chat. See #5730

**Reply by Zarh** [reply]
@pajlada you wrote "If the need arises for speed, we can re-investigate Global Shortcuts" I don't think the issue is the speed but more the confort. The overlay is used in my case mostly in fullscreen (borderless). Sometimes I have a lot of windows opened behind and to get the focus on the chat I must search it with alt + tab among all other windows opened. I think a global shortcut is a must have to interact with the overlay (to send a message or resizing it)

**Reply by pajlada** [reply]
@Zarh My message was related to making the overlay interactable again for resizing it or moving it. Is this something you'd want to do often?

For sending a message without interacting with Chatterino - yeah global shortcuts seem like a reasonable approach. However, I see that as a new feature request, not something I want locked behind the usage of overlay windows.

**Comment by 8thony** [comment]
Scrollbar highlights should not be visible on the overlay

**Reply by Felanbird** [reply]
These are removed if you enable the setting `Hide scrollbar highlights`

**Reply by 8thony** [reply]
Yes, this removes the scrollbar highlights everywhere. 
However, you cannot scroll back in the overlay itself, if you want to go to a highlight, you have to switch to the application and scroll up there.
(The overlay does not match the configured scrollback limit of the main split (set with setting `Split message scrollback limit`)).

**Reply by 8thony** [reply]
- Fixed with #5769 thanks @Nerixyz

**Comment by Holdo** [comment]
@Nerixyz very nice feature.

Could you please consider adding Zoom support from Chatterino settings?
It seems the Overlay is always using Default zoom setting, which creates an inconsistency like this:

![1 4x-zoom](https://github.com/user-attachments/assets/82aff125-dae8-4edb-8827-cdafe64d6a62)
^ the Overlay contents are too small on a 4K screen. There is a workaround to increase font and emote sizes, but it's still not good as everything else still remains small, like user badges, Chatterino buttons, tabs, etc.

**Comment by ibillingsley** [comment]
It should remember the size and position of the overlay window so you don't need to move and resize it every time.

**Comment by ibillingsley** [comment]
It would be nice to have an option to fade out messages after a configurable number of seconds.

That way you can see recent messages without taking up permanent screen space, especially for small channels with low chat activity.

**Comment by orangedudetv** [comment]
would it be possible to have the overlay window be less visible (as in reducing opacity of the overlay) to a configurable amount when the mouse cursor hovers over it? would be neat for when needing to check UIs and other info under it when in click-through mode
